source/common/upstream/outlier_detection_impl.cc

```diff
@@ -75,6 +76,30 @@ void DetectorHostMonitorImpl::putHttpResponseCode(uint64_t response_code) {
   }
 }
 
+void DetectorHostMonitorImpl::putResult(Result result) {
+  Http::Code http_code = Http::Code::InternalServerError;
+
+  switch (result) {
+  case Result::SUCCESS:
+    http_code = Http::Code::OK;
+    break;
+  case Result::TIMEOUT:
+    http_code = Http::Code::GatewayTimeout;
+    break;
+  case Result::CONNECT_FAILED:
+    http_code = Http::Code::ServiceUnavailable;
+    break;
+  case Result::REQUEST_FAILED:
+    http_code = Http::Code::InternalServerError;
+    break;
+  case Result::SERVER_FAILURE:
+    http_code = Http::Code::ServiceUnavailable;
+    break;
+  }
+
+  putHttpResponseCode(enumToInt(http_code));
```

Comment:
Would `DetectorHostMonitorImpl` benefit from being less HTTP centric internally, in addition to the interface changes in this PR?

Reply:
It might make things a bit cleaner, but I don't think it would be very good bang-for-buck for the time it would take.

Reply:
Yeah I would recommend we see how this evolves. I don't think it's terrible right now, especially since we do have explicit HTTP outlier policies. If it starts looking worse over time we could potentially split it into an HTTP and non-HTTP detector or something.